### PR TITLE
[PPD-151] load test reporting analysis

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- k6 test for report analysis getFlow and getFlowList call 

#### Motivation and Context

- test performance under load

#### How Has This Been Tested?

- run in local environment

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: reporting-analysis/load-test/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Load tests for Payments project
```
Comment (pasqualespica) [inline reporting-analysis/load-test/README.md:1]: Reporting project

File: reporting-analysis/load-test/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+2. execute rest call
+3. clears the table from the test records entered
+
+Example command launch on localhost 
+
+```
+k6 run -e BASE_URL=http://localhost:7071 -e FLOW_TABLE_URL=http://127.0.0.1:10002/devstoreaccount1/flows -e ACCESS_SIGNATURE_QUERY_STRING="?sv=2018-03-28&se=2022-04-28T12%3A15%3A36Z&sp=raud&sig=dwGaSA6cXmAk6ZHPrhrg97up9qmvm0W6IAFUAYfKTdA%3D&tn=flows" -e NUM_OF_ORGANIZATIONS=5 -e NUM_FLOW_ID_FOR_ORG=10 reporting-analysis\load-test\src\reporting-analysis_get_flow_list.js
+```
+
+To obtain ACCESS_SIGNATURE_QUERY_STRING from Microsoft Azure Storage Explorer:
```
Comment (pasqualespica) [inline reporting-analysis/load-test/README.md:29]: ok to get SAS via MS storage explore but link https://docs.microsoft.com/en-us/rest/api/storageservices/create-account-sas documentation too.
In this way user musn't download app to obtain it

File: reporting-analysis/load-test/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Call to test the get flow list call:
+1. initializes the table by inserting test records
+2. execute rest call
+3. clears the table from the test records entered
+
+Example command launch on localhost 
+
+```
+k6 run -e BASE_URL=http://localhost:7071 -e FLOW_TABLE_URL=http://127.0.0.1:10002/devstoreaccount1/flows -e ACCESS_SIGNATURE_QUERY_STRING="?sv=2018-03-28&se=2022-04-28T12%3A15%3A36Z&sp=raud&sig=dwGaSA6cXmAk6ZHPrhrg97up9qmvm0W6IAFUAYfKTdA%3D&tn=flows" -e NUM_OF_ORGANIZATIONS=5 -e NUM_FLOW_ID_FOR_ORG=10 reporting-analysis\load-test\src\reporting-analysis_get_flow_list.js
```
Comment (pasqualespica) [inline reporting-analysis/load-test/README.md:26]: add pre-condition section  in the README to say that localhost test makes use of
  https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=docker-hub

File: reporting-analysis/load-test/src/reporting-analysis_get_flow.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import http from 'k6/http';
+import { check } from 'k6';
+import { generateFakeStringArray, randomString } from './modules/helpers.js';
+
+export let options = {
+	summaryTrendStats: ['avg', 'min', 'med', 'max', 'p(95)', 'p(99)', 'p(99.99)', 'count'],
+	stages: [
+		{ duration: '10s', target: 5 }, // simulate ramp-up of traffic from 1 to 50 users over 1 minutes.
```
Comment (pasqualespica) [inline reporting-analysis/load-test/src/reporting-analysis_get_flow.js:8]: comment is different from the actual condifguration ( ex. `10s` vs `1m` )

File: reporting-analysis/load-test/src/reporting-analysis_get_flow_list.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
+import http from 'k6/http';
+import { check } from 'k6';
+import { generateFakeStringArray, randomString } from './modules/helpers.js';
+
+export let options = {
+	summaryTrendStats: ['avg', 'min', 'med', 'max', 'p(95)', 'p(99)', 'p(99.99)', 'count'],
+	stages: [
+		{ duration: '10s', target: 5 }, // simulate ramp-up of traffic from 1 to 50 users over 1 minutes.
```
Comment (pasqualespica) [inline reporting-analysis/load-test/src/reporting-analysis_get_flow_list.js:8]: look at the comment made on previous test ( configuration mismatch vs comment )

**Review by pasqualespica — CHANGES_REQUESTED** [review]
great work 👍

File: reporting-analysis/load-test/README.md (hunk 1 above)
Comment (alessio-acitelli) [reply reporting-analysis/load-test/README.md]: done

File: reporting-analysis/load-test/README.md (hunk 2 above)
Comment (alessio-acitelli) [reply reporting-analysis/load-test/README.md]: done

File: reporting-analysis/load-test/README.md (hunk 3 above)
Comment (alessio-acitelli) [reply reporting-analysis/load-test/README.md:30]: done

File: reporting-analysis/load-test/src/reporting-analysis_get_flow.js (hunk 4 above)
Comment (alessio-acitelli) [reply reporting-analysis/load-test/src/reporting-analysis_get_flow.js]: done

File: reporting-analysis/load-test/src/reporting-analysis_get_flow_list.js (hunk 5 above)
Comment (alessio-acitelli) [reply reporting-analysis/load-test/src/reporting-analysis_get_flow_list.js]: done

File: reporting-analysis/load-test/src/reporting-analysis_get_flow_list.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
+import http from 'k6/http';
+import { check } from 'k6';
+import { generateFakeStringArray, randomString } from './modules/helpers.js';
+
+export let options = {
+	summaryTrendStats: ['avg', 'min', 'med', 'max', 'p(95)', 'p(99)', 'p(99.99)', 'count'],
+	stages: [
+		{ duration: '1m', target: 5 }, // simulate ramp-up of traffic from 1 to 50 users over 1 minutes.
```
Comment (pasqualespica) [inline reporting-analysis/load-test/src/reporting-analysis_get_flow_list.js:8]: 50 users

File: reporting-analysis/load-test/src/reporting-analysis_get_flow.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
+import http from 'k6/http';
+import { check } from 'k6';
+import { generateFakeStringArray, randomString } from './modules/helpers.js';
+
+export let options = {
+	summaryTrendStats: ['avg', 'min', 'med', 'max', 'p(95)', 'p(99)', 'p(99.99)', 'count'],
+	stages: [
+		{ duration: '1m', target: 5 }, // simulate ramp-up of traffic from 1 to 50 users over 1 minutes.
```
Comment (pasqualespica) [inline reporting-analysis/load-test/src/reporting-analysis_get_flow.js:8]: 50 users

**Review by pasqualespica — CHANGES_REQUESTED** [review]

**Review by pasqualespica — APPROVED** [review]
👍

**Pull request merged**